### PR TITLE
Multi section resources

### PR DESCRIPTION
**Pull request by lightsprint09**
Support for Resource with two dimensional array result

### CODE REVIEW

File: DBNetworkStack+Sourcing/ResourceDataProvider.swift [hunk 1]
```diff
@@ -36,16 +33,16 @@ public class ResourceDataProvider<Object>: ArrayDataProviding {
     public var whenStateChanges: ((ResourceDataProviderState) -> Void)?
     
     /// Section Index Titles for `UITableView`. Related to `UITableViewDataSource` method `sectionIndexTitlesForTableView`
-    public let sectionIndexTitles: Array<String>? = nil
+    public let sectionIndexTitles: [String]? = nil
     
     /// The provided data
-    open var contents: Array<Array<Object>> = []
+    open var contents: [[Object]] = []
     /// Describes the current stae of the data provider. Listen for state changes with the `whenStateChanges` callback
```
Comment (dennispost) [inline DBNetworkStack+Sourcing/ResourceDataProvider.swift]: Typo: [...] current state of [...]

File: DBNetworkStack+Sourcing/ResourceDataProvider.swift [hunk 2]
```diff
@@ -63,28 +60,75 @@ public class ResourceDataProvider<Object>: ArrayDataProviding {
...
+    public init(resource: Resource<[[Object]]>, networkService: NetworkServiceProviding,
                 whenStateChanges: @escaping ((ResourceDataProviderState) -> Void)) {
         self.resource = resource
         self.networkService = networkService
         self.whenStateChanges = whenStateChanges
     }
     
     /**
-     Fetches a new resource.
+     Creates an instance which fetches a gives resource and exposes the result as a DataProvider.
```
Comment (dennispost) [inline DBNetworkStack+Sourcing/ResourceDataProvider.swift]: Typo: [...] wich fetches a given resource [...]

File: DBNetworkStack+Sourcing/ResourceDataProvider.swift [hunk 3]
```diff
@@ -63,28 +60,75 @@ public class ResourceDataProvider<Object>: ArrayDataProviding {
...
+     */
+    public init(networkService: NetworkServiceProviding,
+                whenStateChanges: @escaping ((ResourceDataProviderState) -> Void)) {
+        self.resource = nil
+        self.networkService = networkService
+        self.whenStateChanges = whenStateChanges
+    }
+    
+    /**
+     Creates an instance which fetches a gives resource and exposes the result as a DataProvider.
```
Comment (dennispost) [inline DBNetworkStack+Sourcing/ResourceDataProvider.swift]: Typo: [...] wich fetches a given resource [...]

File: DBNetworkStack+Sourcing/ResourceDataProvider.swift [hunk 4]
```diff
@@ -126,8 +171,8 @@ public class ResourceDataProvider<Object>: ArrayDataProviding {
      
      - parameter updates: The updates.
      */
-    private func dataProviderDidChangeContets(with updates: [DataProviderUpdate<Object>]?) {
-        dataProviderDidUpdate?(updates)
-        whenDataProviderChanged?(updates)
+    private func dataProviderDidChangeContets() {
```
Comment (dennispost) [inline DBNetworkStack+Sourcing/ResourceDataProvider.swift]: Typo: dataProviderDidChangeContent

File: DBNetworkStack+SourcingTests/RessourceDataProviderTests.swift [hunk 5]
```diff
@@ -147,9 +166,8 @@ class ResourceDataProviderTests: XCTestCase {
     
     func testOnNetworkRequestCanceledWithNoEmptyData() {
```
Comment (dennispost) [inline DBNetworkStack+SourcingTests/RessourceDataProviderTests.swift]: Same test as in line 154?

**Review by dennispost — CHANGES_REQUESTED** [review]
Please check all your code and comments for 'contents' as there is no plural of content. ;-) http://dictionary.cambridge.org/de/grammatik/britisch-grammatik/content-or-contents

**Review by ahersch — DISMISSED** [review]

**Pull request merged**